atom/common/api/atom_bindings.cc

```diff
@@ -26,6 +28,12 @@ void Crash() {
   static_cast<DummyClass*>(NULL)->crash = true;
 }
 
+void Hang() {
+  // Temporarily access the current thread.
+  base::ThreadRestrictions::ScopedAllowSingleton allow_singleton;
+  base::WaitableEvent(false, false).Wait();
```

Comment:
using it based on usage in `BrowserThread_CurrentlyOn` method. Can just create an indefinite event ?